packages/gestalt/src/Tabs.js

```diff
@@ -73,45 +128,137 @@ function Tab({
           rounding="pill"
           userSelect="none"
         >
-          <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
-            {children}
-          </Text>
-          {indicator === 'dot' && <Circle />}
+          <Flex alignItems="center" gap={2} justifyContent="center">
+            <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
+              {text}
+            </Text>
+
+            {indicator === 'dot' && <Dot />}
+          </Flex>
         </Box>
       </Link>
     </Box>
   );
 }
 
+const TAB_ROUNDING = 2;
+const TAB_INNER_PADDING = 1;
+
+function TabV2({
+  href,
+  indicator,
+  id,
+  index,
+  isActive,
+  onChange,
+  // No longer supported, will be removed when this variant ships
+  // eslint-disable-next-line no-unused-vars
+  size,
```

Comment:
Can't remove this yet because of Flow, but will as part of shipping the new variant